src/SIM/SIMoutput.C

```diff
@@ -756,15 +755,19 @@ bool SIMoutput::writeGlvS2 (const Vector& psol, int iStep, int& nBlock,
           haveAsol = myProblem->evalSol(lovec,*mySol->getScalarSecSol(),Xt);
         else
           haveAsol = myProblem->evalSol(lovec,*mySol->getVectorSecSol(),Xt);
+        if (haveAsol && j == 1)
+          field.resize(lovec.size(), field.cols());
```

Comment:
Then you can also remove line 749 `field.fill(0.0);`which non longer serves any purpose